feat: expose current property name to transform function

pakhuta opened the pull request:
Add the second argument "key" of transformFn function

Close #329

NoNameProvided commented:
Hi @pakhuta! 

Thanks for the PR! I agree you should be able to receive the current property name in the transform. However to be merged can you please make the following changes: 
  - update the signature of the transform to be: `transformFn({ value, key, object, type })` this way we minimize the breaking change as everyone just has to wrap their parameters in a `{}`,
  - remove all changes not related to the code (eg: changelog change, version number)
  - revert the changes you made in the existing tests, but update their signature to the new format
  - add new tests specifically for this use case

pakhuta commented:
@NoNameProvided , I removed a fork a few days ago 🙃, so I'm not able to push changes to this PR. Let's close this one and continue in a new PR https://github.com/typestack/class-transformer/pull/393.

In a new PR I did next:
- updated the signature of the transform function: `transformFn({ value, key, object, type })`
- removed all changes not related to the code (eg: changelog change, version number)
- reverted the changes in the existing tests, but updated their signature to the new format
- added additional checks to an existed test case. I didn't add a new tests because there already was a test case which checks arguments passed to transform function (`value`, `object`, `type`). So, I just added a check that property `key` is passed as a part of arguments.

NoNameProvided commented:
Superset by #393.
